design/yaml_file_errors.md

```diff
@@ -0,0 +1,49 @@
+# YAML File Errors
+
+When the output of a rendered file results in invalid YAML, the resource is not included in base and not deployed.
+Additionally, the file name and error is included in the kots.io/v1beta1.Installer yaml (see the example below).
+The error often references a line number in the rendered content which is inaccessible to the user.
+This proposal attempts to make the rendered output available to the user so that it is easier to debug the issue.
+
+## Goals
+
+- Surface the rendered YAML content to the user when unmarshalling fails
+
+## Non Goals
+
+- Attempt to automate fixing any common errors such as unescaped quotes or line breaks
+
+## Design
+
+Add a new directory at the root of the file tree `errors` that will include the raw content of all rendered output for yaml files that failed unmarshalling.
```

Comment:
As a stretch, it would be nice if when someone viewed the error'd file, we could display the error message(s) in context. Like at least a note at the header of the page saying `line 26: did not find expected key`, but maybe even highlight the line(s) for them, even if they didn't come there from the version history with a deep link to the line.

Reply:
the idea would be that with the additions to the api (file tree and structured kots.io/v1beta1.Installation) the ui team should be able to achieve this or whatever is designed.

Reply:
I'd prefer to be explicit here, and include that stretch goal for the UI in the design. If, for whatever reason, the API doesn't provide enough information for the frontend to display the in-context messages, we should consider changes. If we're lucky and there are no changes, then great.